packages/next/src/build/webpack/config/blocks/base.ts

```diff
@@ -34,6 +34,8 @@ export const base = curry(function base(
   if (ctx.isDevelopment) {
     if (process.env.__NEXT_TEST_MODE && !process.env.__NEXT_TEST_WITH_DEVTOOL) {
       config.devtool = false
+    } else if (process.env.NEXT_RSPACK) {
+      config.devtool = 'source-map'
```

Comment:
why change this?

Reply:
Rspack currently still generates sourcemaps with `file://` locations when using the `eval` devtool -- our current code to reverse maps expects `webpack[-internal]://`.

Similarly, Turbopack always uses separate map files and `file://` locations. Is there a significant performance impact doing so with Rspack?

Reply:
it seems a rspack's bug, i will fix it
Honestly I don't know why webpack choose eval for development by default(performance reason maybe), personally I like debug without sourcemap, so eval is always a trouble when sourcemap disabled

Reply:
Yeah, as far as I know it's for performance reasons. I'm guessing @sokra or @timneutkens could elaborate.

In my experience the separate map file and file uri sources make for a much better debugging experience. That and `eval` is constrained with things like content security policy, etc.

Reply:
yeah I met eval csp problems in formbricks repo

Reply:
![image](https://github.com/user-attachments/assets/e1559100-67b6-4cce-9dfa-a2c40391359d)
that's weird, it seems both rspack and webpack generates same

Reply:
ok i know the reason, the yj-try-ci branch doesn't enable `EvalSourceMapDevToolPlugin` for rspack, so moduleFilenameTemplate not working, I will fix it